src/java.base/share/classes/java/util/concurrent/locks/AbstractQueuedLongSynchronizer.java

```diff
@@ -277,6 +277,26 @@ private static void signalNextIfShared(Node h) {
         }
     }
 
+    /**
+     * Invokes acquire for as long as it throws exceptions
+     * @param node which to reacquire
+     * @param arg the acquire argument
+     */
+    private final void reacquire(Node node, long arg) {
+        for(boolean first = true;;) {
+            try {
+                acquire(node, arg, false, false, false, 0L);
+                break;
+            } catch (Error | RuntimeException ex) {
+                // It's not possible to do much here
+                if (first) {
+                    first = false;
+                    // Log JFR event?
```

Comment:
@DougLea @AlanBateman The second question is whether we should declare a JFR event for this use-case or not.

Reply:
Creating/recording a JFR event would require executing Java code so might be problematic.  We could potentially a a JVM function that could be used for many cases, except SOE.